src/enumerable.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    i = 0
+    selected_arr = []
+    while i < length
+      selected_arr << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    selected_arr
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return false if yield(self[i]) == false || yield(self[i]).nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return true if yield(self[i])
```

Comment:
- [ ] my_none? returns true if the block never returns true for all elements
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern